### ISSUE TITLE
TestResult.getEndMillis() returns 0 for skipped configuration - after upgrading testng to 7.0 +

### ISSUE THREAD

**Issue by soumyathampi** [cbeust/testng#2535]
### TestNG Version
> Note: only the latest version is supported

### Expected behavior
TestResult.getEndMillis() should return the endtime for skipped configuration.

### Actual behavior
TestResult.getEndMillis() returns 0 for skipped configuration.

### Is the issue reproductible on runner?

- [x] Maven

### Test case sample

```java 
package com.logmein.piranha.macdriver.systemtests;

import org.testng.Assert;

import org.testng.annotations.BeforeClass;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;
public class CalculatorTest  {
    /**
     * This is a very basic test that launches Calculator on Mac and performs an addition operation
     * and verifies the result. 
     */
    @Test(description = "Simple calculator test.")
    public void calculatorUiTest() {

    }
    /**
     * Setup.
     */
    @BeforeClass
    public void setup() {
        // Fail this configuration
      
        Assert.assertTrue(false);       
    }
    @BeforeMethod
    public void setup1() {       
       //This configuration will be skipped.       
    }
}
```

While creating report the method TestResult.getEndMillis() returns 0 for method setup1

**Comment by krmahadevan** [comment]
@soumyathampi - What version of TestNG are you working with? Why would a skipped configuration have an `endtime` when it never even ran ?

**Comment by soumyathampi** [comment]
I tried with TestNG 7.3. We add the duration of all methods in our report, which is calculated as StartTime-EndTime. EndTime is supposed to be same as Starttime, if it never ran. That was the behavior before and duration would be 0.

**Comment by krmahadevan** [comment]
@soumyathampi - The latest released version of TestNG is `7.4.0`. Please try that and post back the results.

**Comment by soumyathampi** [comment]
Reproducible on TestNG 7.4.0 also.
